overlord/state/notices.go

```diff
@@ -243,7 +243,12 @@ func (s *State) AddNotice(userID *uint32, noticeType NoticeType, key string, opt
 
 	now := options.Time
 	if now.IsZero() {
-		now = time.Now()
+		now = timeNow()
+		// ensure that two notices never have the same sent time
+		if !now.After(s.noticeLastDate) {
```

Comment:
Given we're comparing equality, why not `if now.Equal(s.noticeLastDate)`?

Reply:
Actually, I think using `!now.After(s.noticeLastDate)` would also solve the non-monotonic time problem. It might result in notices being created in the "future" according to current system time after a synchronization results in system time moving backwards, but after $/Delta$ seconds, everything would be in order and no notice could ever be issued "before" an existing notice.

Reply:
Using equality would fail if there are three notices with the same timestamp. The first one would have `now()` value, and that value would be stored for future comparisons; the next one, also with `now()`, would be equal to the stored value, so `now()+1` would be used and the stored value would be updated. But the third one would also have `now()`, which is smaller than the value stored, so the algorithm would fail. Using `!After()` ensures that it doesn't matter how many notices have the same timestamp.

Reply:
@zyga would this address your concerns with relying on notice timestamps?